### ISSUE TITLE
Problem: Can't differentiate between Dataverse 'bundles' and zip files

### ISSUE THREAD

**Issue by ross-spencer** [archivematica/Issues#108]
By the time a Dataverse transfer reaches the Transfer workflow it becomes difficult to separate [bundles](http://guides.dataverse.org/en/latest/user/dataset-management.html?highlight=bundle) from zip files in a normal workflow using 'Extract Packages'. See below for demonstrations of various transfer layouts as received from Dataverse.

Example 1 (Both zips are bundles): 
```
root@artefactual:/tmp/am-pipeline-data/watchedDirectories/activeTransfers/dataverseTransfer# tree 1128_tabs/
1128_tabs/
├── metadata
│   ├── agents.json
│   └── dataset.json
├── Test 3.zip
└── Test 4.zip
```
Example 2 (zip is a bundle): 
```
root@artefactual:/tmp/am-pipeline-data/watchedDirectories/activeTransfers/dataverseTransfer# tree 115_tab/
115_tab/
├── depress.zip
└── metadata
    ├── agents.json
    └── dataset.json
```
Example 3 (three bundles, the rest are zip files):
```
152_zips/
├── a00000001.freelist
├── a00000001.gdbindexes
├── a00000001.gdbtable
├── a00000001.gdbtablx
├── a00000001.TablesByName.atx
├── a00000002.gdbtable
├── a00000002.gdbtablx
├── a00000003.gdbindexes
├── a00000003.gdbtable
├── a00000003.gdbtablx
├── a00000004.CatItemsByPhysicalName.atx
├── a00000004.CatItemsByType.atx
├── a00000004.FDO_UUID.atx
├── a00000004.freelist
├── a00000004.gdbindexes
├── a00000004.gdbtable
├── a00000004.gdbtablx
├── a00000004.spx
├── a00000005.CatItemTypesByName.atx
├── a00000005.CatItemTypesByParentTypeID.atx
├── a00000005.CatItemTypesByUUID.atx
├── a00000005.gdbindexes
├── a00000005.gdbtable
├── a00000005.gdbtablx
├── a00000006.CatRelsByDestinationID.atx
├── a00000006.CatRelsByOriginID.atx
├── a00000006.CatRelsByType.atx
├── a00000006.FDO_UUID.atx
├── a00000006.freelist
├── a00000006.gdbindexes
├── a00000006.gdbtable
├── a00000006.gdbtablx
├── a00000007.CatRelTypesByBackwardLabel.atx
├── a00000007.CatRelTypesByDestItemTypeID.atx
├── a00000007.CatRelTypesByForwardLabel.atx
├── a00000007.CatRelTypesByName.atx
├── a00000007.CatRelTypesByOriginItemTypeID.atx
├── a00000007.CatRelTypesByUUID.atx
├── a00000007.gdbindexes
├── a00000007.gdbtable
├── a00000007.gdbtablx
├── a0000000a.gdbindexes
├── a0000000a.gdbtable
├── a0000000a.gdbtablx
├── al_ta_on_shp_eng.zip
├── CCG_BSC_Apr2015.KML
├── CCG_BSC_Apr2015.lyr
├── england_ccg_2013.zip
├── England_ccg_2013.zip
├── england_gor_2001.zip
├── england_nhsat_2013.zip
├── england_nhscr_2013.zip
├── England_nhscr_2013.zip
├── GBR_adm0.zip <--- Bundle
├── GBR_adm1.zip <--- Bundle
├── GBR_adm2.zip <--- Bundle
├── GBR_adm_shp.zip
├── gdb
├── gfsa000b11a_e.shp
├── license.txt
├── metadata
│   ├── agents.json
│   └── dataset.json
├── Regions2015GenClipEofEng.zip
├── Regions_December_2015_Generalised_Clipped_Boundaries_in_England.xml
├── Regions_December_2015_Generalised_Clipped_Boundaries_in_England.zip
├── Regions.zip
├── soils.zip
├── TermsAndConditions.html
├── timestamps
├── wales_ol_2001.zip
└── Wales_ol_2001.zip
```
This may be better handled by the Storage Service and we can rely on the validation done by other Dataverse scripts to confirm that extract has happened successfully.

**Comment by ross-spencer** [comment]
Related: https://github.com/artefactual/archivematica/issues/1121

**Comment by ross-spencer** [comment]
A solution is offered here: https://github.com/artefactual/archivematica-storage-service/commit/1c5deaa791b3e686682814195d23d270f3a762df

**Comment by joel-simpson** [comment]
I have tested the dev branch with this fix (which is included in [SS PR #398](https://github.com/artefactual/archivematica-storage-service/pull/398). It is working as I would expect it to....

If A dataset includes tabular data in certain specified formats, Dataverse creates a number of derivative files (a DDI description file, migrations to other tabular data formats and so on). These related files are "bundled" together, so that when a dataset is retrieved using the Dataverse API, the "bundle" is delivered as a ZIP file. 
The storage Service is correctly identifying bundles that are ZIP files, and extracting them in advance. This means that the dataset as it appears to Archivematica includes all of the individual files. 

If a dataset includes a .zip file that isn't a "bundle" of files relating to tabular data, the storage service is NOT extracting it (as we expect). This then leaves the choice up to users to extract packages or not (using the processing configuration).  

This will need final QA once the PR is actually merged, but for now it is looking good.
